Dockerfile

```diff
@@ -1,23 +1,16 @@
 FROM python:3.6-alpine
 
-RUN apk --no-cache add \
-    python3 \
-    git \
-    libpq \
-    wget \
-    libffi \
-    libffi-dev \
-    ca-certificates \
-    python3-dev \
-    postgresql-dev \
-    build-base \
-    bash \
-    curl
-RUN update-ca-certificates
+RUN apk add --update --no-cache libpq postgresql-dev libffi libffi-dev bash libstdc++ nodejs
+RUN apk add --update --no-cache --virtual=build-dependencies build-base 
+RUN apk --repository http://dl-3.alpinelinux.org/alpine/edge/testing/ --update add leveldb leveldb-dev
 
 WORKDIR /app
 ADD requirements.txt .
 RUN pip install -r requirements.txt
+RUN npm install -g os-types
+
+RUN apk del build-dependencies
+RUN rm -rf /var/cache/apk/*
```

Comment:
Can L12-13 be put before `WORKDIR /app` etc, as the results will change less frequently than requirement.txt installation, making better use of the Docker cache and help speed up Docker builds.

Reply:
build-dependencies is needed for the pip install - we install it for installing python libs with native code.

Reply:
Sure, but can these lines me moved before the `WORKDIR /app` line. It just optimises the caching.

Reply:
No - they must come after `pip install -r requirements.txt`, which is after `WORKDIR /app`

They are before the `ADD . .` which is what mostly changes.

Reply:
Gotcha - I'd misunderstood what `del` was doing.